### DISCUSSION TITLE
Como me tornar melhor no Front End

### DISCUSSION THREAD

**Discussion by pietrobelluno** [frontendbr/forum#656] [Discussão]
Boa noite galera !
Queria saber como posso melhorar no front-end , comecei a trabalhar ano passado em uma empresa de desenvolvimento web , comecei programando back-end e quando nosso front-end saiu eu comecei a aprender front para saber o básico e me virar , só que gostei tanto que acabei focando mais nisso do que no back-end hahahah , eu usava bootstrap inicialmente , mas depois de um tempo eu comecei a fazer o CSS puro com minhas próprias classes , mas tenho varias duvidas , tipo queria saber qual a melhor pratica de usar o html , tipo por que usar section article se posso usar divs , qual a diferença sabe? Porque usar SASS ou LESS , eu faço os sites relativamente rápido mesmo sem esses frameworks. Se um bom curso de UI ajuda , por que vi um curso que me interessou mas ele mostra UI meio que na parte de design não em desenvolvimento , basicamente eu quero expandir meu conhecimento , se tiverem dicas de algum framework de CSS interessante que esta sendo bem usado , ou algum tipo de padrão de desenvolvimento no CSS bastante utilizado . Obrigado a todos !

**Comment by pietrobelluno** [comment]
Também tenho uma dúvida sobre gulp , tipo muitas pessoas falam , pra que ele serve ? Onde eu trabalho ninguem usa e ninguem sabe nada a respeito

**Comment by rubenmarcus** [comment]
[comece por aqui](http://letanure.github.io/2015/02/10/por-que-nao-podemos-encontrar-desenvolvedores-front-end/)

ai se souber inglês tem mais alguns que posso passar.
Eu li esse artigo do Tanure, tem 1 ano e meio e ainda to no processo!! Boa sorte!

**Comment by woliveiras** [comment]
Esse artigo é muito bom, @rubenmarcus!!! 🤘 🤘 🤘 

Acredito que a melhor maneira de conseguir o que você está querendo é participando de foruns de discussão e praticando muito, @pietrobp.

Não existe um curso que vai te preparar para todos os cenários e problemas que você vai enfrentar no dia a dia. 

Quando tiver alguma dúvida pesada, conte com a comunidade e taca pau postar sua dúvida aqui no forum.

Artigos que acredito que podem ajudar a direcionar seus estudos:

- [Trilha de estudos Front End](https://woliveiras.com.br/posts/guia-de-estudos-desenvolvedor-front-end-iniciante/)
- [Como se tornar desenvolvedor front end](https://willianjusten.com.br/como-se-tornar-um-desenvolvedor-front-end/)
- Do zero ao herói front end [parte 1](https://www.felipefialho.com/blog/2016/do-zero-a-heroi-do-front-end-parte-1) e [parte 2](https://www.felipefialho.com/blog/2016/do-zero-a-heroi-do-front-end-parte-2)
- [FreeCodeCamp](https://freecodecamp.com)

Da uma olhada também no [carreira front end](https://github.com/woliveiras/front-end-career/).

**Comment by Wmitrut** [comment]
O Jaydson fez uma palestra interessante sobre o assunto. Indico dar uma olhada.

![image](https://user-images.githubusercontent.com/5539257/29214112-6b244830-7e7c-11e7-8d8a-d259097dd41f.png)(https://www.youtube.com/watch?v=LBxHq3IG4F4&ct=t(BrazilJS_Weekly_468_9_2013) )

**Comment by tarciozemel** [comment]
> por que usar section article se posso usar divs , qual a diferença sabe?

Semântica web. Elementos semânticos têm significado e deixam o conteúdo claro (inclusive para robôs); elementos não semânticos não deixam o conteúdo claro.

> Porque usar SASS ou LESS , eu faço os sites relativamente rápido mesmo sem esses frameworks.

Não são frameworks; são pré-processadores CSS.

Eles possuem recursos que estendem as capacidades nativas de CSS; é possível "programar CSS" com loops, condicionais, repetidores, funções, etc.

Ah, e Sass é melhor que Less. xD

> Se um bom curso de UI ajuda, por que vi um curso que me interessou mas ele mostra UI meio que na parte de design não em desenvolvimento.

Ajudar, ajuda, mas provavelmente não vai ser conhecimento essencial para o dia-a-dia.

> Se tiverem dicas de algum framework de CSS interessante que esta sendo bem usado, ou algum tipo de padrão de desenvolvimento no CSS bastante utilizado.

"framework de CSS" tem centenas por aí, como Pure, Bulma, Milligram, Schema, siimple, etc.

De frameworks front-end em geral (não apenas CSS), [também tem sobrando](http://usablica.github.io/front-end-frameworks/compare.html). Recomendo fortemente o Foundation (o que não quer dizer que estou dizendo que deva ser usado em todo e qualquer projeto).

Para "padrão de desenvolvimento no CSS bastante utilizado", isso envolve convenções de nomenclatura, arquitetura CSS e outros. Se quiser dar uma olhadinha, [fiz um curso especificamente sobre isso](https://www.udemy.com/css-um-passo-adiante/).

**Comment by rubenmarcus** [comment]
@woliveiras  Esse artigo do @willianjusten  está bem enxuto,direto e simplificado. Pelo artigo dele eu estou entre intermediário e avançado.

Queria a opinião de vocês, dentro desse tópico aqui, Existe mercado para se chegar num nivel master? E chegar num nivel master sendo remoto/freelancer? A impressão que fica é que em 1 ano ou dois, tudo que ele descreve como Master, ja vai cair pra Avançando e Intermediário e nunca vai se conseguir chegar lá.

OU  que vai acontecer o Apocalipse igual falou o @diegoeis , e o mercado vai enxugar.

Eu tenho muito problema pra aprender coisas novas ,porque no meu trabalho no dia-dia é muito mais do mesmo. Aplicação de frameworks, uma estilização aqui e ali, tratamento de código em JQuery e etc, até em empresas quando trabalhei e sugeri teste de JS e até de CSS (Backstop JS), pessoal falava: nossa que legal, mas ai nada de implementar.

Um caminho é investir em opensource e projetos pessoais é isso?


Essa semana uma startup de Berlin fez uma entrevista comigo e uma das perguntas foi essa: 
- Você participa da comunidade do seu local? Ajuda em Open Source e alguma coisa? Desde que eu li esse artigo do Tanure, eu comecei a sacar que ir nos eventos e contribuir com opensource, não só ia contribuir mais na minha carreira, mas talvez fosse um escape, de ser refém de clientes ou empresas. E como satisfação pessoal também de contribuir um pouco e aprender muito no caminho também, muito mais do que ter marcas e clientes famosos no Portfolio, ou ouvir por exemplo de um amigo que trabalha com SEO e Marketing Digital na Europa: Nossa, mas seu portfolio é muito bom, era pra vc estar ganhando mais de 10mil no Brasil, e aqui na Europa , você encontra emprego fácil etc etc, Mas eu não me deixo desiludir e sei que tem muito que aprender e desenvolver, não só a parte tecnica,mas interpessoal, de comunicação e também todo o resto que envolve em ser um profissional. 

Ou por exemplo da situação que eu ainda ser MEI e ver novatos com menos de 3 anos de carreira, falando que já consegue emitir notas maiores que um MEI, não sinto inveja dessa galera nova que está ai conquistando tudo bem, mas sinto que perdi muito tempo com clientes ou profissionais que não tinham a minima vontade de entender talvez o lado tecnico da parada, ou seja tocar tudo sempre no Go Horse.

**Comment by woliveiras** [comment]
É muito bom você expor essa sua situação dessa maneira, @rubenmarcus.

Com certeza vai ajudar muita gente!

Sobre os tópicos que você questionou eu penso o seguinte: 

> Existe mercado para se chegar num nivel master? E chegar num nivel master sendo remoto/freelancer?

Eu acredito que sim. Porém tenho uma ressalva que é a que falo pra pergunta de baixo.

> A impressão que fica é que em 1 ano ou dois, tudo que ele descreve como Master, ja vai cair pra Avançando e Intermediário e nunca vai se conseguir chegar lá

Tudo o que as pessoas dizem nos seus posts ou em respostas aqui no forum vão sim virar coisa básica daqui a um ou dois anos. Isso é um fato.
Antigamente as pessoas se destacavam no mercado por saber usar pré processadores, por exemplo, hoje isso é parte básica da stack da maioria das empresas.
Antigamente quem sabia Backbone ou Angular tinha o diferencial, hoje é só mais do mesmo (React, Angular, Ember, Vue, etc).
Antigamente quem sabia usar Grunt e montar um bom esquema de build era a melhor pessoa para entrar numa equipe ágil... Hoje...

Isso é normal... Faz parte da evolução do mercado. Imagine que antigamente quem manuseava bem uma tesoura de jardinagem tinha uma produtividade imensa, hoje eles usam carrinhos de controle remoto pra cortar grama. 

Toda profissão tem sua evolução e isso é normal.

Só que quem sabe a base, de verdade, não vai ficar pra trás. Quem sabe computação nunca vai ficar pra trás de alguém que **só** sabe usar framework.

Fora que existe outro ponto importante nisso tudo: o mercado é muito grande. Existem vagas para pessoas que desejam trabalhar apenas com o que algumas pessoas dizem ser o básico e vagas pra o que algumas pessoas costumam chamar de dev modo Exódia. Acredito que é só uma questão de procurar qual é o seu perfil e focar nisso.

* Se for o perfil de mexer muito com UI, foque em ser master nisso.
* Se for o perfil de mexer com programação, foque em ser master nisso.

Por que é muita coisa pra se aprender e muita coisa em que da para virar especialista até mesmo quando já decidimos pra onde ir. Se formos aprender tudo o que aparece de novo na nossa área ficaremos perdidos e patinando sem sair do lugar.

Uma coisa que eu sempre fiz é ver o que o mercado pede e aprender somente aquilo. Não ficar aprendendo de tudo, por que no final das contas é tudo a mesma coisa.

O mesmo que o React faz, o Vue faz. O mesmo que o Grunt faz, o Gulp faz (e agora temos webpack pra abstrair mais algumas coisas). Então se entendermos por baixo dos panos como essas coisas funcionam será muito fácil trocar de uma para a outra.

Um cara que é ninja em ajudar nisso é o @fdaciuk (:heart:), por que nos cursos dele ele ensina como usar uma lib (no caso o React) e explica o que ela faz por baixo dos panos e pode até confirmar se meu ponto de vista é bom.

> O Apocalipse igual falou o @diegoeis , e o mercado vai enxugar

Eu acredito que esse "apocalipse" vai vir sim. E isso não é ruim. Só que vai demorar muito e mesmo que tudo mude sempre teremos vagas, só serão diferentes do que fazemos hoje.

> Um caminho é investir em opensource e projetos pessoais é isso?

Eu estou incomodado como o open source está sendo banalizado hoje em dia por pessoas que buscam fama através de um commit ou projeto novo e acabam desvirtuando um princípio do código aberto que é a **contribuição** com um projeto de outra pessoa (ou para todo um mercado) ou aprendizado (só que isso é conversa para mais de dias e não cabe num comentário 😂), mas sim... O open source é, sim, um caminho para estudar coisas novas e fazer algo fora do nosso contexto normal.

Ex.: alguém que trabalhe somente com jQuery ou Angular pode contribuir em um projeto onde é usado React.

Uma coisa legal de se fazer quando quer aprender algo novo é: pegue um **problema pessoal** (talvez até uma ferramenta que você usa e não gosta de como ela está e resolve o problema) e crie um software que ajude a resolver seu problema com o que você vai estudar.

Isso ajuda a sairmos daqueles exemplos comuns que sempre fazemos em cursos ou tutoriais, que são sempre um caminho único direto a um resultado final já testado por outra pessoa, na net e nos força a aprender mais ainda quebrando a cabeça com o que vai aparecer.

**Comment by rubenmarcus** [comment]
muito bom @woliveiras  !! valeu mesmo

**Comment by didiraja** [comment]
Comece por este livro, te dá toda a base teórica pra voce poder andar com as próprias pernas depois https://www.casadocodigo.com.br/products/livro-guia-frontend
